src/utils/use-onboarding-navigation.ts

```diff
@@ -1,47 +1,56 @@
-import {RootNavigationProps, RootStackParamList} from '@atb/stacks-hierarchy';
+import {RootNavigationProps} from '@atb/stacks-hierarchy';
 
 import {useNavigation, StackActions} from '@react-navigation/native';
 
 import {useCallback} from 'react';
 import {InteractionManager} from 'react-native';
 import {useOnboardingFlow} from './use-onboarding-flow';
+import {useAppState} from '@atb/AppContext';
+import {OnboardingSectionId} from './use-onboarding-sections';
 
 export const useOnboardingNavigation = () => {
   const navigation = useNavigation<RootNavigationProps>();
-  const {getNextOnboardingScreen} = useOnboardingFlow();
+  const {getNextOnboardingSection} = useOnboardingFlow();
+  const {completeOnboardingSection} = useAppState();
 
   const goToScreen = useCallback(
-    (replace, screenName, params?) => {
+    (replace, screen: {name?: any; params?: any}) => {
       InteractionManager.runAfterInteractions(() => {
+        if (!screen?.name) return;
         if (replace) {
-          navigation.dispatch(StackActions.replace(screenName, params));
+          navigation.dispatch(
+            StackActions.replace(screen.name, screen?.params),
+          );
         } else {
-          navigation.navigate(screenName, params);
+          navigation.navigate(screen.name, screen?.params);
```

Comment:
should not be `screen?.name` as `screen.params?`

Reply:
Not in this case because of `if (!screen?.name) return;` on line 19.
Basically params can be defined or undefined, but navigating to a screen with undefined screenName will fail and should never happen.

Reply:
Aha, I read it the other way if (`screen.name` should be `screen?.name`). Updated :+1:

Reply:
The changes are not reflected yet 🤔

Reply:
Maybe you have a commit filter active?
https://github.com/AtB-AS/mittatb-app/pull/4290/commits/b705ec786780bddda62d4494ac912fcab496b365